tests/unit/utils/safeList.js

```diff
@@ -0,0 +1,306 @@
+const xml2js = require('xml2js');
+const { recoverListing } = require('../../../utils/safeList');
+
+
+describe('test safeListObjectVersions', () => {
+    const testCases = [
+        {
+            description: 'should parse empty response',
+            response: {
+                ListVersionsResult: {
+                    Name: 'my-bucket',
+                    IsTruncated: false,
+                    MaxKeys: 1000,
+                },
+            },
```

Comment:
I think a test with an empty response would make sense here, to make sure we respond correctly